packages/languages/src/browser/language-client-contribution.ts

```diff
@@ -106,7 +108,8 @@ export abstract class BaseLanguageClientContribution implements LanguageClientCo
     deactivate(): void {
         this.toDeactivate.dispose();
     }
-    protected doActivate(toDeactivate: DisposableCollection): void {
+    protected async doActivate(toDeactivate: DisposableCollection): Promise<void> {
+        await this.languageServerService.setStartParameters(LanguageContribution.getPath(this), this.getStartParameters());
```

Comment:
It sets parameters only once. Should each language take care of resetting parameters, e.g. if the workspace is changed?